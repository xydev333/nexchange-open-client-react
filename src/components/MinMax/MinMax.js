import React from 'react';
import { I18n } from 'react-i18next';
import isFiatOrder from 'Utils/isFiatOrder';
import styles from './MinMax.scss';

const MinMax = props => {
  if (props.order && isFiatOrder(props.order) && props.type === 'Deposit') return null;

  let clickable = false;
  if(props.setValue){
    clickable = true;
  }

  return (
    <I18n ns="translations">
      {t => (
        <div className={`${styles.minmax} ${props.home ? styles.home : ''}`}>
          <p className={`${props.min > props.amount && props.amount !== '...' ? 'error' : ''} ${clickable ? 'clickable' : ''}`}
            onClick={() => {props.setValue(props.min);}}>
            {t('exchangewidget.min')}: {props.min}
          </p>

          <p className={`${props.max < props.amount && props.amount !== '...' ? 'error' : ''} ${clickable ? 'clickable' : ''}`}
            onClick={() => {props.setValue(props.max);}}>
            {t('exchangewidget.max')}: {props.max}
          </p>
        </div>
      )}
    </I18n>
  );
};

export default MinMax;
