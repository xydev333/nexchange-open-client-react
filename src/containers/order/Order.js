import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchOrder } from '../../actions';

import isFiatOrder from '../../helpers/isFiatOrder';
import config from '../../config';

import '../../css/order.scss';

import OrderInfo from './OrderInfo';
import OrderTop from './OrderTop';
import CoinProcessed from './CoinProcessed';

import NotFound from '../../components/NotFound';
import Loading from '../../components/Loading';

import Notifications from '../../containers/Notifications';
import RefundAddress from '../../containers/RefundAddress';
import ReferralBox from '../../containers/ReferralBox';

class Order extends Component {
  constructor(props) {
    super(props);

    if (this.props.order && this.props.match.params.orderRef === this.props.order.unique_reference) {
      this.state = { order: this.props.order };
    } else {
      this.state = {};
    }
  }

  componentDidMount() {
    this.props.fetchOrder(this.props.match.params.orderRef);
  }

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      clearTimeout(this.timeout);
      clearInterval(this.interval);
      this.props.fetchOrder(this.props.match.params.orderRef);
    }
  }

  componentWillUnmount() {
    clearInterval(this.interval);
    clearTimeout(this.timeout);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order });

    this.timeout = setTimeout(() => {
      this.props.fetchOrder(this.props.match.params.orderRef);
    }, config.ORDER_DETAILS_FETCH_INTERVAL);

    if (nextProps.order !== 429) {
      this.setState({ order: nextProps.order });

      if (
        this.props.order &&
        this.props.order.status_name.length > 0 &&
        this.props.order.status_name[0][0] === 11 &&
        nextProps.order.status_name[0][0] === 12
      ) {
        window.ga('send', 'event', 'Order', 'order paid', nextProps.unique_reference);
      }
    }
  }

  render() {
    if (this.state.order == null) {
      return <Loading />;
    } else if (this.state.order === 404) {
      return <NotFound />;
    } else if (typeof this.state.order === 'object') {
      return (
        <div id="order" className={isFiatOrder(this.state.order) ? 'order-fiat' : 'order-crypto'}>
          <div className="container">
            <OrderTop order={this.state.order} />

            <div className="row">
              <CoinProcessed type="Deposit" order={this.state.order} />
              <CoinProcessed type="Receive" order={this.state.order} />

              <OrderInfo order={this.state.order} />
              <Notifications order={this.state.order} />

              {!isFiatOrder(this.state.order) && <RefundAddress order={this.state.order} />}

              <ReferralBox order={this.state.order} />
            </div>
          </div>
        </div>
      );
    }
  }
}

const mapStateToProps = ({ order }) => ({ order });

export default connect(mapStateToProps, { fetchOrder })(Order);
