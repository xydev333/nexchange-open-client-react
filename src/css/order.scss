@import "variables";
@import "mixins";
@import "bootstrap/variables";
@import "bootstrap/mixins";

#order { background: #f8f7f8; padding: 15px 0 0 0;
	a { color: $green; 
		&:visited, &:focus { text-decoration: none; }
		&:hover { text-decoration: underline; }
		&.btn { color: #fff;
			&:hover { text-decoration: none; }
		}
	}
	#order-ref { margin-bottom: 15px; }
	.box {
		h4 { font-size: 15px; }
		h5, h6 { margin: 0; text-transform: none; }
		.coin-icon { position: relative; top: 8px; font-size: 28px;
			@media (min-width: $screen-sm-min) { top: 9px; }
			@media (min-width: $screen-md-min) { top: 11px; }
		}
		.text-warning { color: $green; }		

		@media (min-width: $screen-sm-min) { margin-bottom: 30px; }
	}

	&.order-fiat {
		.box.fiat {
			h5 { margin-top: 11px; padding-bottom: 7px;
				@media (min-width: $screen-sm-min) { margin-top: 11px; padding-bottom: 13px; }
			}
		}
	}

	.coin-box { padding-left: 10px;
		.tooltip-inner { text-align: left; }
	}
	#time-remaining { color: #2cb4a0; }
	.order-status-section { 
		h2 { margin-top: 0; }
		.explanation { font-size: 16px; margin-top: 15px; }
	}


	@media (min-width: $screen-sm-min) { padding: 25px 0;
		h5 { font-size: 15px; }
		h6 { font-size: 12px; }
		.box h4 { font-size: 18px; }
	}

	@media (min-width: $screen-md-min) {
		h5 { font-size: 17px; }
		h6 { font-size: 16px; }
	}
}

#order-header {
	#bookmark-button { font-size: 20px; padding: 0; margin-top: 0; color: $green; }

	@media (min-width: $screen-ms-min) {
		h3 { float: left; }
		#bookmark-button { float: right; margin-top: 22px; }
	}
}

#order-payment { text-align: center;
	h3 { margin-top: 0; }
	@media (min-width: $screen-ms-min) { text-align: left; }
}

#bookmark {
	.bookmark-text{ font-size: 19px; }
	.input-group-addon { cursor: pointer; padding-top: 0;
		i { font-size: 18px; }
	}
	.form-group { margin-top: 10px; }
	input { font-size: 18px; }
}

.order-status-section {
	h2 { font-size: 24px; }

	@media (min-width: $screen-sm-min) {
		h2 { font-size: 30px; }
	}
}

#share-referral, #notifications { text-align: center;
	.box { padding: 20px; }
	.share { margin-top: 15px; }
	.fa { font-size: 45px; margin: 0 15px 0 0; color: #d6d6d6; }
	h2 { margin-top: 0; font-size: 22px;
		@media (min-width: $screen-sm-min) { font-size: 26px; }
		@media (min-width: $screen-md-min) { font-size: 34px; }
	}
}

#notifications {
	.form-group { margin-top: 10px; }
	button { width: 100%; margin-bottom: 0; }
	form { padding-top: 15px;
		h4 { position: absolute; top: -15px; left: 50%; transform: translate(-50%, 0); }
	}
	@media (min-width: $screen-sm-min) { button { width: auto; } }
}

@include keyframes(color-change) {
	from { color: #adadad; }
	to { color: #fbc02d; } 
}

@import "order-status";
@import "referral";
