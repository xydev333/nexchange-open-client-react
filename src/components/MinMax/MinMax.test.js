import React from 'react';
import { shallow } from 'enzyme';
import orderFiat from 'Mocks/orderFiat';
import order from 'Mocks/order';
import MinMax from './MinMax';

describe('MinMax', () => {
  it('renders correctly correct amounts', () => {
    const wrapShallow = shallow(<MinMax min={1} max={2} amount={1.5} home="true" order={order} type="Deposit" />).dive();
    const wrapShallowHome = shallow(<MinMax min={1} max={2} amount={1.5} order={order} type="Deposit" />).dive();
    const wrapShallowFiatDeposit = shallow(<MinMax min={1} max={2} amount={1.5} order={orderFiat} type="Deposit" />);
    const wrapShallowFiatReceive = shallow(<MinMax min={1} max={2} amount={1.5} order={orderFiat} type="Receive" />).dive();

    expect(wrapShallow).toMatchSnapshot();
    expect(wrapShallowHome).toMatchSnapshot();
    expect(wrapShallowFiatDeposit).toMatchSnapshot();
    expect(wrapShallowFiatReceive).toMatchSnapshot();
  });

  it('renders correctly too low amounts', () => {
    const wrapShallowHome = shallow(<MinMax min={1} max={2} amount={0.5} home="true" order={order} type="Deposit" />).dive();
    const wrapShallowCrypto = shallow(<MinMax min={1} max={2} amount={0.5} order={order} type="Deposit" />).dive();;

    expect(wrapShallowHome).toMatchSnapshot();
    expect(wrapShallowCrypto).toMatchSnapshot();
  });

  it('does not render when order is fiat and type Deposit', () => {
    const wrapShallowFiatTooLow = shallow(<MinMax min={1} max={2} amount={0.5} order={orderFiat} type="Deposit" />);
    const wrapShallowFiatTooHigh = shallow(<MinMax min={1} max={2} amount={3} order={orderFiat} type="Deposit" />);

    expect(wrapShallowFiatTooLow.html()).toBeNull();
    expect(wrapShallowFiatTooHigh.html()).toBeNull();
  });

  it('renders correctly too high amounts', () => {
    const wrapShallowHome = shallow(<MinMax min={1} max={2} amount={3} home="true" order={order} type="Deposit" />).dive();
    const wrapShallowCrypto = shallow(<MinMax min={1} max={2} amount={3} order={order} type="Deposit" />).dive();

    expect(wrapShallowHome).toMatchSnapshot();
    expect(wrapShallowCrypto).toMatchSnapshot();
  });
});
