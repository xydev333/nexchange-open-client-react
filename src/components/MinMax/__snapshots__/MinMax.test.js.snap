// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinMax renders correctly correct amounts 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly correct amounts 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly correct amounts 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MinMax
    amount={1.5}
    max={2}
    min={1}
    order={
        Object {
            "amount_base": "19.85661318",
            "amount_btc": 0.01376142748939,
            "amount_eur": 85.88036048245088,
            "amount_quote": "100.00000000",
            "amount_quote_fee": 9,
            "amount_usd": 100,
            "created_on": "2028-05-26T18:27:22.310989Z",
            "deposit_address": null,
            "display_refund_address": false,
            "from_default_rule": false,
            "is_default_rule": true,
            "kyc_deadline": "2028-05-26T19:27:52.086814Z",
            "pair": Object {
              "base": Object {
                "code": "NANO",
                "has_enabled_pairs": true,
                "has_enabled_pairs_for_test": true,
                "is_base_of_enabled_pair": true,
                "is_base_of_enabled_pair_for_test": true,
                "is_crypto": true,
                "is_quote_of_enabled_pair": false,
                "is_quote_of_enabled_pair_for_test": true,
                "maximal_amount": "300.00000000",
                "min_confirmations": 1,
                "minimal_amount": "0.01000000",
                "name": "nano",
                "withdrawal_fee": "0.10000000",
              },
              "fee_ask": 0.04,
              "fee_bid": 0.04,
              "name": "NANOUSD",
              "quote": Object {
                "code": "USD",
                "has_enabled_pairs": true,
                "has_enabled_pairs_for_test": true,
                "is_base_of_enabled_pair": false,
                "is_base_of_enabled_pair_for_test": false,
                "is_crypto": false,
                "is_quote_of_enabled_pair": true,
                "is_quote_of_enabled_pair_for_test": true,
                "maximal_amount": "1000.00000000",
                "min_confirmations": null,
                "minimal_amount": "50.00000000",
                "name": "USD",
                "withdrawal_fee": "0.00000000",
              },
            },
            "payment_deadline": "2028-05-26T18:42:22.310989Z",
            "payment_url": "",
            "payment_window": 15,
            "price": Object {
              "rate": 4.55989196,
              "rate_btc": 0.000503655,
              "rate_eur": 3.12975487,
              "rate_usd": 3.64443412,
            },
            "referral_code": Array [
              Object {
                "code": "RB1EZWMKQLH",
                "created_on": "2028-05-11T13:09:46.202181Z",
                "modified_on": "2028-05-11T13:09:46.202198Z",
              },
            ],
            "status_name": Array [
              Array [
                11,
                "INITIAL",
              ],
            ],
            "transactions": Array [
              Object {
                "address_to": 12215,
                "amount": "19.85661318",
                "confirmations": 1,
                "created_on": "2028-05-26T18:28:58.010666Z",
                "currency": "NANO",
                "is_completed": true,
                "is_verified": true,
                "modified_on": "2028-05-26T18:31:27.333333Z",
                "time": "2028-05-26T18:28:58.010506Z",
                "tx_id": "21A6F4E385B0374C023D2F2BAA846BBAA65C7C3093B8324DD4A13A06F4652E7F",
                "type": "W",
              },
            ],
            "unique_reference": "OCFP7W",
            "user_provided_amount": 0,
            "withdraw_address": Object {
              "address": "xrb_1fqajjogym4dwr8uoi9rt34cpx4qyswh35tib6g11i4k18kosha4z5xr1yz6",
              "currency_code": "NANO",
              "destination_tag": null,
              "name": "",
              "payment_id": null,
              "type": "W",
            },
            "withdrawal_fee": 0.1,
            "withdrawal_fee_quote": 0.455989196,
          }
    }
    type="Deposit"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly correct amounts 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too high amounts 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className="error"
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className="error"
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too high amounts 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className="error"
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className="error"
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too low amounts 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="error"
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="error"
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too low amounts 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="error"
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="error"
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
